### ISSUE TITLE
Missing Merge Conflicts on > 5000 Changes

### ISSUE THREAD

**Issue by michvllni** [microsoft/vscode#117040]
Issue Type: <b>Bug</b>

1. Merge something with more than 5000 changed files
2. Have merge conflicts in several files, including files not in the first 5000 changed files
3. VS Code only shows the ones in the first 5000 changed files

The preferred behaviour would be that VS Code shows all conflicts, probably by reducing the number of viewed staged changes

VS Code version: Code 1.53.2 (622cb03f7e070a9670c94bae1a45d78d7181fbd4, 2021-02-11T11:48:04.245Z)
OS version: Windows_NT x64 10.0.17763

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz (4 x 2800)|
|GPU Status|2d_canvas: unavailable_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: disabled_off<br>protected_video_decode: disabled_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|undefined|
|Memory (System)|31.99GB (18.79GB free)|
|Process Argv|--crash-reporter-id 0ce7feb5-248f-4de5-9392-61843a35929d|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (4)</summary>

Extension|Author (truncated)|Version
---|---|---
beautify|Hoo|1.5.0
jbockle-format-files|jbo|3.0.0
al|ms-|6.5.413786
powershell|ms-|2020.6.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspyt653:30253241
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstry244:30256636
pythonvsdeb440:30248342
pythonvsded773:30248341
pythonvspyt600cf:30251589
pythontb:30258531

```

</details>

**Comment by joaomoreno** [comment]
Yeah we currently hardcode changes to 5000 max. We could potentially increase this by now.

**Comment by michvllni** [comment]
I think for this issue it would be enough to show the conflicts first, then show the staged files afterwards.
With a hard coded limit there might always be the case that the changed files superceed it.
Maybe make it changeable through the settings?
And show something like "5000+" on the files to indicate it is more than 5000

**Comment by kodkuce** [comment]
Just head same issue, can this be please set in settings so people dont have to build form source
